Packs/Syslog/Integrations/SyslogSender/SyslogSender.py

```diff
@@ -0,0 +1,280 @@
+    'LOG_USER': SysLogHandler.LOG_USER,
+    'LOG_UUCP': SysLogHandler.LOG_UUCP,
+    'LOG_LOCAL0': SysLogHandler.LOG_LOCAL0,
+    'LOG_LOCAL1': SysLogHandler.LOG_LOCAL1,
+    'LOG_LOCAL2': SysLogHandler.LOG_LOCAL2,
+    'LOG_LOCAL3': SysLogHandler.LOG_LOCAL3,
+    'LOG_LOCAL4': SysLogHandler.LOG_LOCAL4,
+    'LOG_LOCAL5': SysLogHandler.LOG_LOCAL5,
+    'LOG_LOCAL6': SysLogHandler.LOG_LOCAL6,
+    'LOG_LOCAL7': SysLogHandler.LOG_LOCAL7
+}
+SEVERITY_DICT = {
+    'Unknown': 0,
+    'Low': 1,
+    'Medium': 2,
+    'High': 3,
+    'Critical': 4
+}
+
+''' Syslog Manager '''
+
+
+class SyslogManager:
+    def __init__(self, address: str, port: int, protocol: str, logging_level: int, facility: int):
+        """
+        Class for managing instances of a syslog logger.
+        :param address: The IP address of the syslog server.
+        :param port: The port of the syslog server.
+        :param protocol: The messaging protocol (TCP / UDP).
+        :param logging_level: The logging level.
+        """
+        self.address = address
+        self.port = port
+        self.protocol = protocol
+        self.logging_level = logging_level
+        self.facility = facility
+
+    @contextmanager  # type: ignore[misc, arg-type]
+    def get_logger(self) -> Logger:
+        """
+        Get a new instance of a syslog logger.
+        :return: syslog logger
+        """
+        handler = self._get_handler()
+        syslog_logger = self._init_logger(handler)
+        try:
+            yield syslog_logger
+        finally:
+            syslog_logger.removeHandler(handler)
+            handler.close()
+
+    def _get_handler(self) -> SysLogHandler:
+        """
+        Get a syslog handler for a logger according to provided parameters.
+        :return: syslog handler
+        """
+        address: Union[str, Tuple[str, int]] = (self.address, self.port)
+        kwargs: Dict[str, Any] = {
+            'facility': self.facility
+        }
+
+        if self.protocol == 'tcp':
+            kwargs['socktype'] = SOCK_STREAM
+        elif self.protocol == 'unix':
+            address = self.address
+
+        kwargs['address'] = address
+
+        return SysLogHandler(**kwargs)
+
+    def _init_logger(self, handler: SysLogHandler) -> Logger:
+        """
+        Initialize a logger with a syslog handler.
+        :param handler: A syslog handler
+        :return: A syslog logger
+        """
+        syslog_logger = getLogger('SysLogLogger')
+        syslog_logger.setLevel(self.logging_level)
+        syslog_logger.addHandler(handler)
+
+        return syslog_logger
+
+
+''' HELPER FUNCTIONS '''
+
+
+def init_manager(params: dict) -> SyslogManager:
+    """
+    Create a syslog manager instance according to provided parameters.
+    :param params: Parameters for the syslog manager.
+    :return: syslog manager
+    """
+    address = params.get('address', '')
+    port = int(params.get('port', 514))
+    protocol = params.get('protocol', 'udp').lower()
```

Comment:
In line 136 you are using uppercase UDP. Need to be consistent. Probably best to use constants.